source/fuzz/protobufs/spvtoolsfuzz.proto

```diff
@@ -1225,6 +1226,21 @@ message TransformationOutlineFunction {
 
 }
 
+message TransformationOutlineSelectionConstruct {
+
+  // Creates a new selection construct with a header in |new_header_block_id|
+  // and a merge block in |new_merge_block_id|.
+
+  // Result id of the OpLabel instruction. The instruction belongs to the block
+  // that will be converted to the header block of a new selection construct.
+  uint32 new_header_block_id = 1;
+
+  // Result id of the OpLabel instruction. The instruction belongs to the block
+  // that will be a merge block of a new selection construct.
+  uint32 new_merge_block_id = 2;
+
+}
```

Comment:
I don't think the description is clear enough.  From reading it, I don't know quite what the transformation does.

I'm also not sure about the transformation name.

I suggest naming it "TransformationWrapRegionInSelection".

I would call the fields "region_entry_block_id", "region_exit_block_id".  I would also add another bool field called "branch_condition", so that the fuzzer pass can randomly decide whether the branch condition should be true or false (and set up the branch targets accordingly).

For the description I would say something like:

"Takes a single-entry-single-exit region R, and turns it into either:

if(true) { R }

or

if(false) { } else { R }

The entry block for R becomes a selection header and the exit block a selection merge.
"

I would then comment the fields as being "The entry block for the region R", and "The exit block for the region R".

Reply:
Fixed in 44c64b8.